src/Microsoft.VisualBasic/tests/VB/FileIOTests.vb

```diff
@@ -0,0 +1,901 @@
+
+            ' Ensure copy didn't overwrite existing data
+            Using stream As New IO.StreamReader(IO.File.OpenRead(testFileDest))
+                Dim readData(SourceData.Length - 1) As Char
+                stream.Read(readData, 0, SourceData.Length)
+                Assert.Equal(DestData, readData)
+            End Using
+            TestBase.Dispose()
+        End Sub
+
+        <Fact>
+        Public Sub CopyFile_FileSourceFileName_DestinationFileName_OverwriteTrue()
+            Dim TestBase As New FileIOTestBase
+            Dim testFileSource As String = TestBase.GetTestFilePath()
+            Dim testFileDest As String = TestBase.GetTestFilePath()
+
+            ' Write and copy file
+            Using sourceStream As New IO.StreamWriter(IO.File.Create(testFileSource))
+                Using destStream As New IO.StreamWriter(IO.File.Create(testFileDest))
+                    sourceStream.Write(SourceData, 0, SourceData.Length)
+                    destStream.Write(DestData, 0, DestData.Length)
+                End Using
+            End Using
+            FileSystem.CopyFile(testFileSource, testFileDest, overwrite:=True)
+
+            ' Ensure copy transferred written data
+            Using stream As New IO.StreamReader(IO.File.OpenRead(testFileDest))
+                Dim readData(SourceData.Length - 1) As Char
+                stream.Read(readData, 0, SourceData.Length)
+                Assert.Equal(SourceData, readData)
+            End Using
+            TestBase.Dispose()
+        End Sub
+        <Fact(Skip:="Not Implemented")>
+        Public Sub CopyFile_SourceFileName_DestinationFileName_UIOption()
+
+        End Sub
+
+        <Fact(Skip:="Not Implemented")>
+        Public Sub CopyFile_SourceFileName_DestinationFileName_UIOption_UICancelOption()
+
+        End Sub
+
+        <Fact>
+        Public Sub CreateDirectory_Directory()
+            Dim TestBase As New FileIOTestBase
+            Dim FullPathToNewDirectory As String = IO.Path.Combine(TestBase.TestDirectory(), "NewDirectory")
+            Assert.False(IO.Directory.Exists(FullPathToNewDirectory))
+            FileSystem.CreateDirectory(FullPathToNewDirectory)
+            Assert.True(IO.Directory.Exists(FullPathToNewDirectory))
+            TestBase.Dispose()
+        End Sub
+
+        <Fact>
+        Public Sub CurrentDirectoryGet()
+            Dim CurrentDirectory As String = IO.Directory.GetCurrentDirectory()
+            Assert.Equal(FileSystem.CurrentDirectory, CurrentDirectory)
+        End Sub
+
+        <Fact>
+        Public Sub CurrentDirectorySet()
+            Dim SavedCurrentDirectory As String = IO.Directory.GetCurrentDirectory()
+            Dim TestBase As New FileIOTestBase
+            FileSystem.CurrentDirectory = TestBase.TestDirectory
+            Assert.Equal(FileSystem.CurrentDirectory, TestBase.TestDirectory)
+            FileSystem.CurrentDirectory = SavedCurrentDirectory
```

Comment:
why we are testing the same thing twice ? what id currentDirectory = testDirectory ?, show that currentDirectory is not equal to the setDirectoryPath

Reply:
We aren't I save CurrentDirectory because I have no idea what it might be
, then change it to something I know and verify it changed with assert. Then I put it back to what it was before I started and verify it is correct.